# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/JSContext.h
# Commit: 954c1651adb2
# Full Hash: 954c1651adb203c44392e127d6487aec3de53c97
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2020-07-07 09:47:47
# Regressor Bug: 1646378
# File Overlap Count: 1
# Description:
#   Bug 1646378: Add CallInlinedFunction r=jandem
#   
#   This is mostly the same as CallScriptedFunction. The key differences are:
#   
#   1. We have an extra argument, the ICScript, which must be stored in the JSContext so the Baseline prologue can find it.
# ==============================================================================

diff -r 01851025fa25 -r 954c1651adb2 js/src/vm/JSContext.h
--- a/js/src/vm/JSContext.h	Thu Jul 02 23:02:23 2020 +0000
+++ b/js/src/vm/JSContext.h	Mon Jul 06 19:05:21 2020 +0000
@@ -467,6 +467,10 @@
   }
 #endif
 
+  static size_t offsetOfInlinedICScript() {
+    return offsetof(JSContext, inlinedICScript_);
+  }
+
  public:
   js::InterpreterStack& interpreterStack() {
     return runtime()->interpreterStack();
